### ISSUE TITLE
S1172 false positive in Visual Studio 2022 IntelliSense

### ISSUE THREAD

**Issue by tonto7973** [SonarSource/sonar-dotnet#5553]
### Description

S1172 unused parameter reported in VS 2022 when the parameter is used. This is coming from Visual Studio IntelliSense only. Running a build does not produce these errors.

### Repro steps

Sample code
```csharp
/*
Warning S1172 Remove this parameter 'value', whose value is ignored in the method.
Unused parameters are misleading. Whatever the values passed to such parameters, the behavior will be the same.
*/
private static dynamic ToDynamic(object value)
{
    var json = JsonConvert.SerializeObject(value); // <-- parameter 'value' used here
    return JsonConvert.DeserializeObject(json, typeof(ExpandoObject));
}
```

### Expected behavior

S1172 not reported

### Actual behavior

S1172 reported

### Known workarounds

Suppress this warning, e.g.:

```
dotnet_diagnostic.S1172.severity = none
```

### Related information

IDE:
- Visual Studio Enterprise 2022 Version 17.1.3
- SonarLint Version 6.1.0.45943
- OS Windows 10

Project:
- SonarAnalyzer.CSharp Version 8.37.0.45539
- net6.0

**Comment by doterik** [comment]
#5491

**Comment by pavel-mikula-sonarsource** [comment]
We're investigating under #5491,thanks @doterik
